src/lib/utils/hooks/useGetInfoFromIds.ts

```diff
@@ -82,6 +82,9 @@ export default function useGetInfoFromIds<T extends EntityHeader | UserProfile>(
   const idProp = (type: HookType) =>
     type === 'USER_PROFILE' ? 'ownerId' : 'id'
```

Comment:
The problem with hooks is that there are restrictions e..g they can't be used conditionally or inside of the callback.
Long term I think we need to store entity/user mapping in context or just local storage. But that's a refactor for another day. For now -- this is somewhat hacky but if we know that there is another component that's using the same entities as the component that uses the hook this seems to work. (for now)

Reply:
I have some amount of concern with session storage, I think it mostly comes from having not used it before. But like you said we'll eventually get around to having a react context/store like way of saving these entities.